### ISSUE TITLE
Layout bug on iOS 8.3

### ISSUE THREAD

**Issue by extremer** [chiahsien/CHTCollectionViewWaterfallLayout#94]
Today, our app is released to the Appstore and layout is buggy just to show every cell in wrong position.
I want to make sure that it's my fault on usage or bug inside the library.

Is anyone experiencing this problem?
Everything's just okay when running the app on lower(< 8.3) version of simulator.
Thanks in advance.

**Comment by extremer** [comment]
update:
It only happens on iphone5, iphone5s(couldn't test on iphone5c).
Everything works find over iphone6 so I thinks the resolution is the thing matters.

**Comment by extremer** [comment]
I found the problem.
When I addsubview to collectionViewCell's contentView, contentView doesn't stick to collectionViewCell and it has wrong position(wrong origin, right size). Therefore I set the following code to collectionViewCell's initWithFrame

if( SYSTEM_VERSION_FROM(@"8.0") )
{
    [self.contentView setTranslatesAutoresizingMaskIntoConstraints:FALSE];
}
else
{
    [self.contentView setAutoresizingMask:UIViewAutoresizingFlexibleHeight|UIViewAutoresizingFlexibleHeight];
}

and it worked before iOS8.3.

When I remove 'setTranslatesAutoresizingMaskIntoConstraints:FALSE' part, layout seems right but a bunch of layout constraint warning comes up.
I'm curious how do other people do when adding subviews to contentView without problem.

**Comment by extremer** [comment]
Solved.

I guess it's iOS8.3 bug.
Someone says there's same thing happening for UITableView.
http://stackoverflow.com/questions/29572043/ios-8-3-uitableview-cell-not-aligned-indentation-to-the-left

Similarly, I added following code in initWithFrame and everything works fine.
Hope it helps someone.

[self addConstraint:[NSLayoutConstraint constraintWithItem:self.contentView attribute:NSLayoutAttributeLeading relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeLeading multiplier:1.0f constant:0.0f]];
[self addConstraint:[NSLayoutConstraint constraintWithItem:self.contentView attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeTop multiplier:1.0f constant:0.0f]];
